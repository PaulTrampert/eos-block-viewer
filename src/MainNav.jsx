import React from 'react';
import PropTypes from 'prop-types';
import {
  Navbar,
  Button
} from 'react-bootstrap';

function MainNav({onRequestBlocks}) {
  return (
    <Navbar bg="dark" variant="dark" className="justify-content-between">
      <Navbar.Brand>EOS Block Viewer</Navbar.Brand>
      <Button onClick={onRequestBlocks}>Load</Button>
    </Navbar>
  );
}

MainNav.propTypes = {
  onRequestBlocks: PropTypes.func.isRequired
}

export default MainNav;