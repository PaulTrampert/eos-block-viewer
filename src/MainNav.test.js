import React from 'react';
import {shallow} from 'enzyme';
import {Button} from 'react-bootstrap';
import MainNav from './MainNav';


describe('MainNav', () => {
  let onRequestBlocks;
  let subject;

  beforeEach(() => {
    onRequestBlocks = jest.fn();
    subject = shallow(<MainNav onRequestBlocks={onRequestBlocks} />);
  });

  describe('render', () => {
    it('renders the nav bar containing the site title and load button', () => {
      expect(subject).toMatchSnapshot();
    });

    it("binds onRequestBlocks to the load button's onClick event", () => {
      expect(subject.find(Button).props().onClick).toBe(onRequestBlocks);
    });

    describe('when blocks are loading', () => {
      beforeEach(() => {
        subject.setProps({
          loadingBlocks: true
        });
      });

      it('the load button is disabled', () => {
        expect(subject).toMatchSnapshot();
      });
    });
  });
});