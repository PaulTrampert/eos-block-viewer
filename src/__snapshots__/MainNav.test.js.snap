// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNav render renders the nav bar containing the site title and load button 1`] = `
<Navbar
  bg="dark"
  className="justify-content-between"
  collapseOnSelect={false}
  expand={true}
  variant="dark"
>
  <NavbarBrand>
    EOS Block Viewer
  </NavbarBrand>
  <Button
    active={false}
    disabled={false}
    onClick={[MockFunction]}
    type="button"
    variant="primary"
  >
    Load
  </Button>
</Navbar>
`;
